### PR TITLE
Better invariants - always route list/tuple to their requisite VTs instead of ConstantVariable

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109869

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Comment by ezyang** [comment]
Can you do the ConstantVariable.create change on its own? I'll stamp it. (BTW, if you don't want to update the call sites, you can technically make the constructor do what you want by defining `__new__` on ConstantVariable lol)

**Comment by voznesenskym** [comment]
> Can you do the ConstantVariable.create change on its own? I'll stamp it. (BTW, if you don't want to update the call sites, you can technically make the constructor do what you want by defining `__new__` on ConstantVariable lol)

no `__new__` please. Also `.create` is a well known pattern in dynamo. Also sure, can make the change on its own.

**Review by jansel — APPROVED** [review]

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change, but the failed test looks legit as it is also failing in trunk https://hud.pytorch.org/pytorch/pytorch/commit/06aa6966a88586d34e6470cc2149121d17971056' -c nosignal

Let me check why it's classified as flaky (it shouldn't)

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -161,14 +161,14 @@ def load_triton():
...
-        return roi_align is not None and hasattr(
-            getattr(torch.ops, "torchvision", None), "roi_align"
-        )
-    except ImportError:
-        return False
+    #     return roi_align is not None and hasattr(
+    #         getattr(torch.ops, "torchvision", None), "roi_align"
+    #     )
+    # except ImportError:
+    return False
```
Comment (peterbell10) [inline torch/_inductor/utils.py:171]: Staged the wrong file?

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: torch/_inductor/utils.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_inductor/utils.py:171]: yeah, weird

**Comment by voznesenskym** [comment]
The CI fiasco on this pr has been vaguely awful

**Pull request closed without merging**